packages/api/src/base/Decorate.ts

```diff
@@ -344,8 +344,15 @@ export default abstract class Decorate<ApiType extends ApiTypes> extends Events
     decorated.key = (arg1?: Arg, arg2?: Arg): string =>
       u8aToHex(compactStripLength(creator(creator.meta.type.isDoubleMap ? [arg1, arg2] : arg1))[1]);
 
-    decorated.keyPrefix = (): string =>
-      u8aToHex(creator.keyPrefix);
+    decorated.keyPrefix = (key1?: Arg): string =>
```

Comment:
The problem here is that we now have duplication of code, as per https://github.com/polkadot-js/api/blob/master/packages/api/src/base/Decorate.ts#L466-L473

So from a maintenance pov we need to ensure that we maintain exactly the same in 2 places. This is problematic.

Reply:
Can extract this into a function call.